proposals/drafts/campaign_finance_filings.rst

```diff
@@ -0,0 +1,298 @@
+    the basic process of filing a document with an agency.
+
+Rationale
+=========
+
+Political committees file statements with their regulators in order to disclose
+funders, expenditures, organizational ties and other financial details.
+Filings encapsulate some number of possible specific disclosures, and some
+filings can contain more than one type of disclosure, so this proposed option is
+to consider each actual filing as an object and then each specific disclosed
+piece of information as associated with that object. So a CommitteeNameChange
+would be associated with a Filing, along with a CommitteeAddressChange and a
+CommitteeOfficerAddition. A series of CommitteeContributions and
+CommitteeExpenses would be attached to one Filing. This way we don't need to
+model the specific disclosure forms in each state, and just directly extract
+the most meaningful parts of the filing.
+
+We want to make it easy for people to do manipulations on top of this data,
+while ensuring that this system itself does as little manipulation as possible
+to the underlying data from reports filed. For instance, many records include
+Person entities - these are deterministically extracted from just the available
+filing data, e.g. an address, employer, occupation and name are all extracted
+for each contribution. Deduplication can then happen with each of these
+entities, but they are themselves totally reproducible from just the filing
+information.
+
+This model entails that Filings have Sections - a Filing can, and often will,
+have multiple Sections. As much of the specific logic as possible should be
+pushed down into Sections, which offer flexibility by allowing us to compose
+them together to reproduce the data from state-specific reports. We can model
+a particular state filing form as a Filing with a specific set of expected
+Sections to be present. Some attributes that might make sense as properties of a
+Filing, such as Amendments, are here presented as Sections; the line between the
+two is muddy, but this proposal tries to keep Filings as minimal as possible.
+
+Finally, the different entity types presented here present either current or
+likely future conflicts with other entities in the Open Civic Data namespace, so
+this proposal doesn't make any attempt to resolve those conflicts since they
+seem ancillary for now.
+
+TODO
+----
+* Make filing activity actions use same pattern as OCD bill actions (@aepton)
+
+Questions to answer
+-------------------
+* How to model Committee Status and Committee Purpose inside of OCD Organization
+  models?
+* How to reconcile multiple reports that describe the same contribution,
+  expenditure or other event?
+* How to handle different types of Committees beyond just imputing their types
+  based on (say) whether they are oriented toward more than one Candidate?
+
+Implementation
+==============
+
+Filing
+------
+
+id
+    Open Civic Data-style id in the format ``ocd-cf-filing/{{uuid}}``
```

Comment:
It is not specified in any OCD spec (AFAIK). The defacto reference implementation of OCD right now is pupa, and it uses [uuid.uuid1()](https://docs.python.org/3.4/library/uuid.html#uuid.uuid1)

https://github.com/opencivicdata/pupa/blob/274cd0ddf9d550c5b20cd99d6d19720041d21222/pupa/scrape/base.py#L151